### DISCUSSION TITLE
Scaling extra mixed extra features

### DISCUSSION THREAD

**Discussion by pkardy** [google/lightweight_mmm#187] [General]
How would I scale extra features if they contain boolean values like holidays and discrete values like price? 

currently Im dividing by the mean :
`scaler_control = preprocessing.CustomScaler(divide_operation=jnp.mean)`

The problem appears with the binary data when predicting, since holiday is 0 and it yields nan's. 

Also,

why is the there a multiplication in the cost scaler?
`scaler_cost = preprocessing.CustomScaler(divide_operation=jnp.mean, multiply_by=0.15)`

**Comment by steven-struglia** [comment]
I'm not sure you need to scale binary or categorical features (I never have in any context). The regression just turns that variable into an indicator function and calls it a day. 

Regarding the `multiply_by`, that seems to just be a scaling hyperparameter to the model. As far as I have tested, it's important to tune that parameter to your data in order to get a model that makes theoretical and intuitive sense in terms of how the contribution plot, model fit, and ROI estimates play out. I personally used it to ensure that baseline organic attribution / KPI didn't disappear from the regression by way of a vanishing regression coefficient since my media spends are quite large. It's just a matter of playing around with it and seeing what works, then sticking with that and moving through the rest of the analysis process.
